spinand.c

```diff
@@ -0,0 +1,279 @@
+}
+
+static int spi_nand_wait_for_page(struct flashctx *flash, unsigned int row_addr)
+{
+	int ret;
+
+	ret = spi_nand_read_page(flash, row_addr);
+	if (ret)
+		return ret;
+	return spi_nand_wait(flash);
+}
+
+static int spi_nand_read_cache(struct flashctx *flash, unsigned int column_addr,
+                               uint8_t *bytes, unsigned int len)
+{
+	uint8_t cmd[1 + SPINAND_COLUMN_ADDR_LEN] = { SPINAND_READ_CACHE, };
+
+	int addr_len = spi_nand_prepare_column_address(cmd, column_addr);
+
+	/* Send Read */
+	return spi_send_command(flash, 1 + addr_len + DUMMY_BYTE, len, cmd, bytes);
+}
+
+static int spi_nand_inpage_read(struct flashctx *flash, unsigned int row_addr, unsigned int column_addr,
+                                uint8_t *bytes, unsigned int len, unsigned int chunksize)
+{
+	/* Read data within one page */
+	int ret;
+
+	if (chunksize == 0)
+		chunksize = len;
+	ret = spi_nand_wait_for_page(flash, row_addr);
+	if (ret)
+		return ret;
+
+	while (len) {
+		unsigned int data_to_read = min(chunksize, len);
+		ret = spi_nand_read_cache(flash, column_addr, bytes, data_to_read);
+		if (ret)
+			return ret;
+		len -= data_to_read;
+		column_addr += data_to_read;
+		bytes += data_to_read;
+		printf("page %d\n",len);
+	}
+
+	return 0;
+}
+
+static inline unsigned int ilog2(unsigned int x) {
+	int res = 0;
+	while (x >>= 1)
+		++res;
+	return res;
+}
+
+int spi_nand_read_chunked(struct flashctx *flash, uint8_t *buf, unsigned int start,
+                          unsigned int len, unsigned int chunksize)
+{
+	/* Read data over multiple pages */
+	int ret;
+
+	/* Calculate acture column address len */
+	unsigned int column_addr_len = ilog2(flash->chip->page_size);
+	unsigned int column_addr_mask = ~(-1 << column_addr_len);
+	printf("%d %d %d\n",flash->chip->page_size,column_addr_len,column_addr_mask);
+
+	while (len) {
+		unsigned int data_to_read = min(chunksize, len);
+		ret = spi_nand_inpage_read(flash, start >> column_addr_len, start & column_addr_mask, buf, len, data_to_read);
+		if (ret)
+			return ret;
+		len -= data_to_read;
+		start += data_to_read;
+		buf += data_to_read;
+		printf("chunked %d\n",len);
+	}
+
+	return 0;
+}
+
+/***************************************
+ * PROGRAM CYCLE
+ ***************************************/
+
+
+/***************************************
+ * PROBE
+ ***************************************/
+
+static char *strncpy_tospace(char *dest, const char *src, unsigned int len)
+{
+	int i;
+
+	for (i = 0; i < len; i++) {
+		dest[i] = src[i];
+		if (dest[i] == ' ') {
+			dest[i] = '\0';
+			break;
+		}
+	}
+	return dest;
+}
+
+static void probe_spi_nand_toshiba(struct flashctx *flash, const uint8_t *parameters)
+{
+	char *vendor = (char *)malloc(12);
+	char *name = (char *)malloc(20);
+
+	flash->chip->vendor = strncpy_tospace(vendor, (const char *)(parameters + 32), sizeof(vendor));
+	flash->chip->name = strncpy_tospace(name, (const char *)(parameters + 44), sizeof(name));
```

Comment:
`sizeof(vendor)` and `sizeof(name)` are getting the size of a pointer which will be architecture dependent. 

Replace the malloc size and this usage size with a `#define`.

Reply:
BTW I know this PR is pretty much abandoned but I'm using it as a starting point for a new PR supporting the ONFI gold spec. Consider these notes to myself if you don't plan on revisiting.

Reply:
Oh, nice to see someone would like to introduce spi nand support finally. I'm glad to help you if you need (if I can still recall why and how I wrote this...)

Reply:
Hi! Thanks for putting this PR out! I'll reach out if I run into any unknowns. Planning to write my own implementation and coming back to this PR to compare if I run into issues. Glad to see you're still around and active!